src/datatypes/src/vectors/helper.rs

```diff
@@ -572,14 +574,27 @@ mod tests {
         assert_eq!(&array, &vector.to_arrow_array());
     }
 
+    #[test]
+    fn test_try_binary_array_into_vector() {
+        let input_vec = vec!["hello".as_bytes(), "world".as_bytes()];
+
+        let input_large_binary_array: ArrayRef =
+            Arc::new(LargeBinaryArray::from(input_vec.clone()));
+        let assertion_array: ArrayRef = Arc::new(BinaryArray::from(input_vec));
+        let vector = Helper::try_into_vector(input_large_binary_array.clone()).unwrap();
+
+        assert_eq!(2, vector.len());
+        assert_eq!(0, vector.null_count());
+
+        let output_arrow_array: ArrayRef = vector.to_arrow_array();
+        assert_eq!(&DataType::Binary, output_arrow_array.data_type());
+        assert_eq!(&assertion_array, &output_arrow_array);
```

Comment:
I didn't put any assertions for `ArrowDataType::Binary`(not sure if this test case is useful)  and `ArrowDataType::FixedSizeBinary(_)` (come across problems to convert it to `ArrayRef`)

Please let me know if those 2 cases should be added?

Reply:
A better way may be creating a BinaryVector from input for assertion. We convert large_binary_array, binary_array and fixed_binary_array into output vectors and then assert output vectors with the expected `BinaryVector`.